File: VB/ApplicationServerService/ApplicationServerService.vb
Imports System
Imports System.Configuration
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Diagnostics
Imports System.ServiceProcess
Imports System.Text
Imports DevExpress.ExpressApp
Imports DevExpress.ExpressApp.Security
Imports DevExpress.Persistent.Base
Imports DevExpress.ExpressApp.Security.ClientServer
Imports System.Runtime.Remoting
Imports DevExpress.ExpressApp.Security.ClientServer.Remoting
Imports DevExpress.ExpressApp.Xpo
Imports DevExpress.ExpressApp.MiddleTier
Imports System.ServiceModel
Imports DevExpress.ExpressApp.Security.Strategy
Imports DevExpress.ExpressApp.Security.ClientServer.Wcf

Namespace ApplicationServerService
    Partial Public Class ApplicationServerService
        Inherits System.ServiceProcess.ServiceBase

        Private Sub serverApplication_DatabaseVersionMismatch(ByVal sender As Object, ByVal e As DatabaseVersionMismatchEventArgs)
            e.Updater.Update()
            e.Handled = True
        End Sub
        Private serviceHost As ServiceHost
        Protected Overrides Sub OnStart(ByVal args() As String)
            ValueManager.ValueManagerType = GetType(MultiThreadValueManager(Of )).GetGenericTypeDefinition()
            'string connectionString = ConfigurationManager.ConnectionStrings["ConnectionString"].ConnectionString;
            InMemoryDataStoreProvider.Register()
            Dim connectionString As String = InMemoryDataStoreProvider.ConnectionString
            Dim serverApplication As New ApplicationServerServiceServerApplication()
            serverApplication.ConnectionString = connectionString
            serverApplication.Setup()
            serverApplication.CheckCompatibility()
            serverApplication.Dispose()
            Dim securityProviderHandler As QueryRequestSecurityStrategyHandler = Function() New SecurityStrategyComplex(GetType(SecuritySystemUser), GetType(SecuritySystemRole), New AuthenticationStandard())
            Dim dataServer As New SecuredDataServer(connectionString, XpoTypesInfoHelper.GetXpoTypeInfoSource().XPDictionary, securityProviderHandler)
            serviceHost = New ServiceHost(New WcfSecuredDataServer(dataServer))
            serviceHost.AddServiceEndpoint(GetType(IWcfSecuredDataServer), WcfDataServerHelper.CreateDefaultBinding(), "http://localhost:1451/DataServer1")
            serviceHost.Open()
        End Sub
        Protected Overrides Sub OnStop()
            serviceHost.Close()
        End Sub
        Public Sub New()
            InitializeComponent()
        End Sub
    End Class
End Namespace

File: CS/ApplicationServerService/ApplicationServerService.cs
using System;
using System.Configuration;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Diagnostics;
using System.ServiceProcess;
using System.Text;
using DevExpress.ExpressApp;
using DevExpress.ExpressApp.Security;
using DevExpress.Persistent.Base;
using DevExpress.ExpressApp.Security.ClientServer;
using System.Runtime.Remoting;
using DevExpress.ExpressApp.Security.ClientServer.Remoting;
using DevExpress.ExpressApp.Xpo;
using DevExpress.ExpressApp.MiddleTier;
using System.ServiceModel;
using DevExpress.ExpressApp.Security.Strategy;
using DevExpress.ExpressApp.Security.ClientServer.Wcf;

namespace ApplicationServerService {
    public partial class ApplicationServerService : System.ServiceProcess.ServiceBase {
        private void serverApplication_DatabaseVersionMismatch(object sender, DatabaseVersionMismatchEventArgs e) {
            e.Updater.Update();
            e.Handled = true;
        }
        private ServiceHost serviceHost;
        protected override void OnStart(string[] args) {
            ValueManager.ValueManagerType = typeof(MultiThreadValueManager<>).GetGenericTypeDefinition();
            //string connectionString = ConfigurationManager.ConnectionStrings["ConnectionString"].ConnectionString;
            InMemoryDataStoreProvider.Register();
            string connectionString = InMemoryDataStoreProvider.ConnectionString;
            ApplicationServerServiceServerApplication serverApplication = 
                new ApplicationServerServiceServerApplication();
            serverApplication.ConnectionString = connectionString;
            serverApplication.Setup();
            serverApplication.CheckCompatibility();
            serverApplication.Dispose();
            QueryRequestSecurityStrategyHandler securityProviderHandler = delegate() {
                return new SecurityStrategyComplex(typeof(SecuritySystemUser), 
                    typeof(SecuritySystemRole), new AuthenticationStandard());
            };
            SecuredDataServer dataServer = new SecuredDataServer(
                connectionString, XpoTypesInfoHelper.GetXpoTypeInfoSource().XPDictionary,
                        securityProviderHandler);
            serviceHost = new ServiceHost(new WcfSecuredDataServer(dataServer));
            serviceHost.AddServiceEndpoint(typeof(IWcfSecuredDataServer),
                WcfDataServerHelper.CreateDefaultBinding(),
                "http://localhost:1451/DataServer");
            serviceHost.Open();
        }
        protected override void OnStop() {
            serviceHost.Close();
        }
        public ApplicationServerService() {
            InitializeComponent();
        }
    }
}
